frontend/packages/dev-console/src/components/import/import-submit-utils.ts

```diff
@@ -407,6 +408,101 @@ export const createOrUpdateDeploymentConfig = (
     : k8sCreate(DeploymentConfigModel, deploymentConfig, dryRun ? dryRunOpt : {});
 };
 
+export const createDevfileResources = async (
+  formData: GitImportFormData,
+  dryRun: boolean,
+  appResources: AppResources,
+  generatedImageStreamName: string = '',
+): Promise<K8sResourceKind[]> => {
+  const verb: K8sVerb = 'create';
+
+  const {
+    name,
+    project: { name: namespace },
+    application: { name: applicationName },
+    git: { url: repository, ref = 'master' },
+    devfile: { devfileSuggestedResources },
+    image: { tag: selectedTag },
+  } = formData;
+
+  const defaultLabels = getAppLabels({ name, applicationName, selectedTag });
+  const defaultAnnotations = {
+    ...getGitAnnotations(repository, ref),
+    isFromDevfile: 'true',
+  };
+
+  const devfileResourceObjects: DevfileSuggestedResources = Object.keys(
+    devfileSuggestedResources,
+  ).reduce((acc: DevfileSuggestedResources, resourceType: string) => {
+    const resource: K8sResourceKind = devfileSuggestedResources[resourceType];
+    return {
+      ...acc,
+      [resourceType]: {
+        ...resource,
+        metadata: {
+          ...resource.metadata,
+          annotations: {
+            ...defaultAnnotations,
+            ...resource.metadata?.annotations,
+          },
+          name,
+          namespace,
+          labels: {
+            ...defaultLabels,
+            ...resource.metadata?.labels,
+          },
+        },
+      },
+    };
+  }, {} as DevfileSuggestedResources);
+
+  return Promise.all([
+    // createOrUpdateImageStream(
+    //   formData,
+    //   devfileResourceObjects.imageStream,
+    //   dryRun,
+    //   appResources,
+    //   verb,
+    //   generatedImageStreamName,
+    // ),
```

Comment:
Dont we need this? The error is related to image stream not found.

Reply:
Yes we do, but that would be where my problem was when I described it on the chat 😄 

Still trying to figure it out -- didn't mean to leave this commented out.